### ISSUE TITLE
testing policy document recommends the wrong group of assert* methods

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4204]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4204|
|Type|defect|
|Created|2010-01-10 23:04:28Z|

In [#2631](https://github.com/twisted/twisted/issues/2631) we added a recommendation that `assertEqual` be preferred in Twisted tests, over `assertEquals`.  This is the opposite of the conclusion I remember the group coming to (although I only remember a count of existing uses as the basis for this decision).

Am I mis-remembering?  Do we really prefer `assertEqual` now?  I don't think so.  We should fix the document to reflect reality.

Attachments:

* [test-doc.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5c5/5c5af838ea967e22c520d771182a34823d000131/bc0afc0f773eca2eb65d8e7f3f206f88a092ba67.patch) (1332 bytes) - added by exarkun on 2010-01-11 03:12:38Z - 
* [enumerated-test-doc.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5c5/5c5af838ea967e22c520d771182a34823d000131/5b86c709f098a41f8fb06738602867bd50c37faa.patch) (1757 bytes) - added by exarkun on 2010-01-11 03:23:18Z - version that avoids "plural" and "singular"

<details><summary>Searchable metadata</summary>

```
trac-id__4204 4204
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1263164668000000 1263164668000000
changetime__1263181424000000 1263181424000000
version__None None
owner__ 
cc__glyph cc__radix cc__itamarst cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

```
18:14 < radix> exarkun: btw you're right we prefer assertEquals
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|spiv commented|
|-|-|

As noted on IRC, "plural" vs. "singular" is both confusing and grammatically incorrect here.

Are there any assert* methods other than assertEqual(s) that we are trying to talk about in that text about plural vs. singular?  I don't think so.

Just say, "... should use the assert forms rather than the fail forms.  Also, assertEquals should be used instead of assertEqual.  assertTrue is also preferred ..."

That's shorter, clearer, and also not nonsense.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Thanks!

There's also `assertNotEquals` and `assertAlmostEquals`. :/  I don't think there are any more, though, so I could just enumerate those.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|spiv commented|
|-|-|

Ok, so the rule is just about -Equal/-Equals variants.  So say that.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|spiv commented|
|-|-|

...or you could enumerate, I guess, sure.  Either way is fine with me.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [27951]) Correct backwards policy on the use of `assertEqual` vs `assertEquals`


Author: exarkun
Reviewer: spiv
Fixes: [#4204](https://github.com/twisted/twisted/issues/4204)

Previously the testing standard mandated `assertEqual` over `assertEquals`.  This
is backwards from the decision we actually came to.  Change the policy document to
suggest `assertEquals` and discourage the use of `assertEqual`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
